### DISCUSSION TITLE
Xaml style setter doesn't work

### DISCUSSION THREAD

**Discussion by n2-holden** [unoplatform/uno#10788] [Q&A] [answered]
Is there something special I need to do to get Styles to work?  I have a simple Style:

```xaml
        <Grid.Resources>
            <Style TargetType="AppBarButton" x:Name="MenuStyle">
                <Setter Property="Foreground" Value="White" />
            </Style>
        </Grid.Resources>
```

Then, in the Grid I have: 

```xaml
        <AppBarButton Icon="World" Label="Companies" BorderThickness="2" Click="Click_Dashboard" Tag="Companies" />
```

The Foreground property doesn't get set in any of the AppBarButton.  I also tried explicitly specifying a style as a StaticResource, but that doesn't even compile.

**Comment by jeromelaban** [comment, marked as answer]
If you want to change styles you have to options, the first with implicit styles (without `x:Name` or `x:Key`), the second with explicit style (with an `x:Name` or `x:Key`). You can have more details in [Microsoft's WinUI documentation about styles](https://learn.microsoft.com/en-us/windows/apps/design/style/xaml-styles).

Here sample you can experiment with: https://playground.platform.uno/#9c23d023

```xaml
<ContentControl>
    <FrameworkElement.Resources>
        <Style TargetType="AppBarButton">
            <Setter Property="Foreground" Value="Orange" />
        </Style>
        <Style TargetType="AppBarButton" x:Key="OtherStyle">
            <Setter Property="Foreground" Value="Red" />
        </Style>
    </FrameworkElement.Resources>

    <StackPanel>
        <AppBarButton Icon="World" Label="Companies" BorderThickness="2" Tag="Companies" />
        <AppBarButton Icon="World" Label="Others" BorderThickness="2" Tag="Companies" Style="{StaticResource OtherStyle}" />
    </StackPanel>
</ContentControl>
```
